strawberryfields/engine.py

```diff
@@ -485,6 +490,22 @@ def run(self, program, *, args=None, compile_options=None, run_options=None):
         eng_run_keys = ["eval", "session", "feed_dict", "shots"]
         eng_run_options = {key: temp_run_options[key] for key in temp_run_options.keys() & eng_run_keys}
 
+        # check that batching is not used together with shots > 1
+        if self.backend_options.get("batch_size", 0) and eng_run_options["shots"] > 1:
+            raise NotImplementedError("Batching cannot be used together with multiple shots.")
+
+        # check that post-selection and feed-forwarding is not used together with shots > 1
+        for p in program_lst:
+            for c in p.circuit:
+                try:
+                    if c.op.select and eng_run_options["shots"] > 1:
+                        raise NotImplementedError("Post-selection cannot be used together with multiple shots.")
+                except AttributeError:
+                    pass
+
+                if c.op.measurement_deps and eng_run_options["shots"] > 1:
+                    raise NotImplementedError("Feed-forwarding of measurements cannot be used together with multiple shots.")
```

Comment:
This is very nice and quite elegant 🤔 I wouldn't have though of using `measurement_deps`.

However, I wonder if this might add some overhead for large programs - perhaps at some point we might want to add properties to the program that return `True` for feedforward/postselection.

Reply:
Might be better to make an issue for this, and do it in a new PR.

Reply:
I was thinking the same. This was the best way I could figure out how to this early in `engine.py` though (and it seemed to work well).